src/components/TodoItem.tsx

```diff
@@ -0,0 +1,98 @@
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+
+interface Props {
+  todo: Todo;
+  toggleCompleted: (a: number, b: boolean) => void;
+  updatingId: number | null;
+  handleSave: (a: number) => void;
+  updatingText: string;
+  setUpdatingText: (a: string) => void;
+  handleEdit: (a: number, b: string) => void;
+  handleDelete: (a: number) => void;
+  temp: Todo | null;
+  isDeleting: boolean;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  toggleCompleted,
+  updatingId,
+  handleSave,
+  updatingText,
+  setUpdatingText,
+  handleEdit,
+  handleDelete,
+  temp,
+  isDeleting,
+}) => {
+  const todoItemClass = classNames('todo', {
+    completed: todo.completed,
+  });
+
+  return (
+    <div key={todo.id} data-cy="Todo" className={todoItemClass}>
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+      <label
+        className="todo__status-label"
+        onClick={() => toggleCompleted(todo.id, todo.completed)}
+      >
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          readOnly
+        />
+      </label>
+
+      {updatingId === todo.id ? (
+        <form
+          onSubmit={e => {
+            e.preventDefault();
+            handleSave(todo.id);
```

Comment:
same as in prev comment with handleAdd.
In general handleSave can live in TodoItem component and not need todo.id as argument.